src/main/java/com/amazon/opendistroforelasticsearch/ad/task/ADTaskManager.java

```diff
@@ -413,14 +527,15 @@ public void stopDetector(
                 ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
                 ADTask adTask = ADTask.parse(parser, searchHit.getId());
 
-                if (!isADTaskEnded(adTask) && lastUpdateTimeExpired(adTask)) {
+                if (adTask.isHistoricalTask() && !isADTaskEnded(adTask) && lastUpdateTimeExpired(adTask)) {
```

Comment:
This if statement will only apply to the historical task?

Reply:
Yes, will add reseting logic for realtime job in next PR as we will change the profile API. Will add TODO to make it clear.

Reply:
When the system is busy, it may stall  historical detectors' search or write longer than 2 piece intervals (10 seconds).  When the system finally manages to run historical detectors' requests, what would historical detector do after you set the state to stopped?  I only found you check if the task is cancelled in ADBatchTaskRunner.checkIfADTaskCancelled, but didn't find you check for stopped state.  Anything I am missing?